Framework/Algorithms/src/GroupToXResolution.cpp

```diff
@@ -0,0 +1,160 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidAlgorithms/GroupToXResolution.h"
+
+#include "MantidAPI/HistogramValidator.h"
+#include "MantidAPI/WorkspaceHasDxValidator.h"
+#include "MantidDataObjects/Workspace2D.h"
+#include "MantidDataObjects/WorkspaceCreation.h"
+#include "MantidHistogramData/HistogramBuilder.h"
+#include "MantidKernel/BoundedValidator.h"
+#include "MantidKernel/CompositeValidator.h"
+
+#include <boost/math/special_functions/pow.hpp>
+
+namespace {
+namespace Prop {
+std::string const FRACTION{"FractionOfDx"};
+std::string const INPUT_WS{"InputWorkspace"};
+std::string const OUTPUT_WS{"OutputWorkspace"};
+} // namespace Prop
+constexpr double FWHM_GAUSSIAN_EQUIVALENT{0.68};
+} // namespace
+
+namespace Mantid {
+namespace Algorithms {
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(GroupToXResolution)
+
+/// Algorithms name for identification. @see Algorithm::name
+const std::string GroupToXResolution::name() const {
+  return "GroupToXResolution";
+}
+
+/// Algorithm's version for identification. @see Algorithm::version
+int GroupToXResolution::version() const { return 1; }
+
+/// Algorithm's category for identification. @see Algorithm::category
+const std::string GroupToXResolution::category() const {
+  return "Transforms\\Rebin";
+}
+
+/// Algorithm's summary for use in the GUI and help. @see Algorithm::summary
+const std::string GroupToXResolution::summary() const {
+  return "Groups points within intervals given by the Dx into single points";
+}
+
+/** Initialize the algorithm's properties.
+ */
+void GroupToXResolution::init() {
+  auto inputValidator = boost::make_shared<Kernel::CompositeValidator>();
+  inputValidator->add(boost::make_shared<API::WorkspaceHasDxValidator>());
+  constexpr bool acceptHistograms{false};
+  inputValidator->add(
+      boost::make_shared<API::HistogramValidator>(acceptHistograms));
+  declareProperty(
+      Kernel::make_unique<API::WorkspaceProperty<>>(
+          Prop::INPUT_WS, "", Kernel::Direction::Input, inputValidator),
+      "An input workspace with Dx values.");
+  declareProperty(Kernel::make_unique<API::WorkspaceProperty<>>(
+                      Prop::OUTPUT_WS, "", Kernel::Direction::Output),
+                  "The grouped workspace.");
+  auto betweenZeroAndOne =
+      boost::make_shared<Kernel::BoundedValidator<double>>(0., 1.);
```

Comment:
I don't see why this has to be capped by 1. One might want to group based on multiples of the resolution.